libraries/CMakeLists.txt

```diff
@@ -12,9 +12,25 @@ macro( clone_path path )
     endif()
 endmacro()
 
+# Macro utility to checkout a submodule that has the update flag set to none
```

Comment:
Possible to update the existing `clone_path` macro by taking an extra parameter to indicate whether `--checkout` flag should be used? 
If the caller does not provide the extra parameter, then the default behavior of cloning can be without the `--checkout` flag

Reply:
Will add this.

Reply:
Added optional parameter to command string